File: support_tools/swigwin-3.0.12/Examples/test-suite/python/template_inherit_runme.py
from template_inherit import *
a = FooInt()
b = FooDouble()
c = BarInt()
d = BarDouble()
e = FooUInt()
f = BarUInt()

if a.blah() != "Foo":
    raise ValueError

if b.blah() != "Foo":
    raise ValueError

if e.blah() != "Foo":
    raise ValueError

if c.blah() != "Bar":
    raise ValueError

if d.blah() != "Bar":
    raise ValueError

if f.blah() != "Bar":
    raise ValueError

if c.foomethod() != "foomethod":
    raise ValueError

if d.foomethod() != "foomethod":
    raise ValueError

if f.foomethod() != "foomethod":
    raise ValueError

if invoke_blah_int(a) != "Foo":
    raise ValueError

if invoke_blah_int(c) != "Bar":
    raise ValueError

if invoke_blah_double(b) != "Foo":
    raise ValueError

if invoke_blah_double(d) != "Bar":
    raise ValueError

if invoke_blah_uint(e) != "Foo":
    raise ValueError

if invoke_blah_uint(f) != "Bar":
    raise ValueError

File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/template_inherit_runme.lua
require("import")	-- the import fn
import("template_inherit")	-- import lib into global
ti=template_inherit --alias

-- catching undefined variables
local env = _ENV -- Lua 5.2
if not env then env = getfenv () end -- Lua 5.1
setmetatable(env, {__index=function (t,i) error("undefined global variable `"..i.."'",2) end})


fi = ti.FooInt()
assert(fi:blah() == "Foo")
assert(fi:foomethod() == "foomethod")

bi = ti.BarInt()
assert(bi:blah() == "Bar")
assert(bi:foomethod() == "foomethod")

assert(ti.invoke_blah_int(fi) == "Foo")
assert(ti.invoke_blah_int(bi) == "Bar")

bd = ti.BarDouble()
success = pcall(ti.invoke_blah_int, bd)
assert(not success)
